# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: browser/components/shell/nsWindowsShellService.cpp
# Commit: 5492a8c99f4a
# Full Hash: 5492a8c99f4aeae1b9bbc4024fb03e97a42809f6
# Author: Ben Hearsum <bhearsum@mozilla.com>
# Date: 2022-05-10 03:50:31
# Description:
#   Bug 1767589: check for S_FALSE where appropriate in PinCurrentAppToTaskbar r=rkraesig
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D145652
# ==============================================================================

diff -r f6dc9b5e231a -r 5492a8c99f4a browser/components/shell/nsWindowsShellService.cpp
--- a/browser/components/shell/nsWindowsShellService.cpp	Mon May 09 13:47:59 2022 +0000
+++ b/browser/components/shell/nsWindowsShellService.cpp	Mon May 09 13:53:46 2022 +0000
@@ -1066,7 +1066,7 @@
   hr = shellWindows->FindWindowSW(&loc, &empty, SWC_DESKTOP, &hwnd,
                                   SWFO_NEEDDISPATCH,
                                   getter_AddRefs(dispDesktop));
-  if (FAILED(hr)) return NS_ERROR_FAILURE;
+  if (FAILED(hr) || hr == S_FALSE) return NS_ERROR_FAILURE;
 
   RefPtr<IServiceProvider> servProv;
   hr = dispDesktop->QueryInterface(IID_IServiceProvider,
